Load PDB debug info (path, guid, age) from PE/COFF object files

ronaldfw opened the pull request:
We add loading of PDB debug info (path, guid, age) to our PE/COFF
object file class so that we can uniquely match PDB files to COFF files.

Tested: Unit tests.
Bug: https://b/194497774

dimitry- commented on src/ObjectUtils/CoffFile.cpp:
```diff
@@ -120,6 +122,31 @@ uint64_t CoffFileImpl::GetExecutableSegmentOffset() const {
+
+  // If 'this' was successfully created with CreateCoffFile, 'object_file_' cannot be nullptr.
+  CHECK(object_file_ != nullptr);
+  if (!object_file_->getDebugPDBInfo(debug_info, pdb_file_path) && debug_info != nullptr) {
```
reverse the if to use early return? Also, we need to consume the llvm::Error, otherwise it will abort().

dimitry- commented on src/ObjectUtils/CoffFile.cpp:
```diff
@@ -120,6 +122,31 @@ uint64_t CoffFileImpl::GetExecutableSegmentOffset() const {
 bool CoffFileImpl::IsElf() const { return false; }
 bool CoffFileImpl::IsCoff() const { return true; }
 
+std::optional<PdbDebugInfo> CoffFileImpl::GetDebugPdbInfo() const {
```
Since getDebugPDBInfo can return a error consider making this ErrorMessageOr?

ronaldfw replied:
Done.

dimitry- commented on src/ObjectUtils/CoffFileTest.cpp:
```diff
@@ -78,3 +79,18 @@ TEST(CoffFile, FileDoesNotExist) {
+  ASSERT_THAT(coff_file_or_error, HasNoError());
+
+  std::optional<PdbDebugInfo> pdb_debug_info = coff_file_or_error.value()->GetDebugPdbInfo();
+  ASSERT_TRUE(pdb_debug_info.has_value());
```
would it be possible to add a test for missing debug info?

ronaldfw replied:
Done.

beckerhe reviewed (approved):
LGTM

ronaldfw commented on src/ObjectUtils/CoffFile.cpp:
```diff
@@ -120,6 +122,31 @@ uint64_t CoffFileImpl::GetExecutableSegmentOffset() const {
+  // If 'this' was successfully created with CreateCoffFile, 'object_file_' cannot be nullptr.
+  CHECK(object_file_ != nullptr);
+  if (!object_file_->getDebugPDBInfo(debug_info, pdb_file_path) && debug_info != nullptr) {
+    if (debug_info->PDB70.CVSignature == kPdb70Signature) {
```
Done.

ronaldfw commented:
@dimitry- Thanks for the review. I think I addressed all your comments. PTAL. I force pushed after rebasing on main due to the llvm revert. Hope this doesn't make it too hard to track the change.

@beckerhe Thanks for the review!

dimitry- reviewed (approved):
thanks!
